### PR TITLE
Style edit form buttons using bootstrap

### PR DESCRIPTION
**Pull request by jeffreyfultonca**
Style buttons, add cancel button, refactor coffeescript accordingly, modify home_page.rb to click button instead of link, add bin/ directory to .gitignore

### CODE REVIEW

**Comment by dbernar1** [comment]
A flaky test! How exciting! Sometimes is passes, sometimes it fails.

**Comment by dbernar1** [comment]
Looks like the flaky test is unrelated to this piece of code.

**Comment by amirci** [comment]
Why do we need to add /bin to .gitignore ?

**Comment by jeffreyfultonca** [comment]
Perhaps we do not need to add /bin to .gitignore. I added it because it looked like all the other repos didn't have a /bin directory and I did. Thoughts?

**Comment by dbernar1** [comment]
What's in your bin?

**Comment by jeffreyfultonca** [comment]
screen shot 2014-03-15 at 7 34 42 pm

**Comment by dbernar1** [comment]
Weird - I wonder how you got that there. Do you remember running "bundle install --binstubs", or the term binstubs at all?

**Comment by swalberg** [comment]
Bin shouldn't be ignored. Binstubs go there and should be checked in.

Sean

On Saturday, March 15, 2014, Dan Bernardic notifications@github.com wrote:

> What's in your bin?
> 
> ## 
> 
> Reply to this email directly or view it on GitHubhttps://github.com/winnipegrb/winnivote/pull/34#issuecomment-37743931
> .

## 

Sean Walberg sean@ertw.com    http://ertw.com/

**Comment by jeffreyfultonca** [comment]
I removed /bin from .gitignore, amended the previous commit with 'commit --amend', then force pushed with 'push -f'. Is that workflow appropriate?

**Comment by swalberg** [comment]
Yes, the git work was correct. In the future please try and keep a pull request on a single topic though -- checking in the binstubs has nothing to do with your CSS/JS work and it should go in its own commit/PR (sorry, I didn't see that the first time around as I was checking on my phone)

File: app/assets/javascripts/ideas.js.coffee [hunk 1]
```diff
@@ -1,19 +1,34 @@
...
+    $("#idea-#{id} .idea-form").fadeOut 'fast', =>
+      $("#idea-#{id} .idea-content").fadeIn 'fast', =>
+
+        title = $("#idea-#{id} .idea-content").find('.title').text()
+        description = $("#idea-#{id} .idea-content").find('.description').text()
+
+        $("#idea-#{id} .idea-form").find('#idea_title').val(title)
+        $("#idea-#{id} .idea-form").find('#idea_description').val(description)
+
+  getId = (button_clicked) -> $(button_clicked).closest('.idea').attr('data-target')
```
Comment (swalberg) [inline app/assets/javascripts/ideas.js.coffee:34]: :green_heart:

File: app/assets/javascripts/ideas.js.coffee [hunk 2]
```diff
@@ -1,19 +1,34 @@
...
+
+    id = getId(this)
+    
+    $("#idea-#{id} .idea-form").fadeOut 'fast', =>
+      $("#idea-#{id} .idea-content").fadeIn 'fast', =>
+
+        title = $("#idea-#{id} .idea-content").find('.title').text()
+        description = $("#idea-#{id} .idea-content").find('.description').text()
+
+        $("#idea-#{id} .idea-form").find('#idea_title').val(title)
```
Comment (swalberg) [inline app/assets/javascripts/ideas.js.coffee:31]: Did you mean to find by ID? It should be unique to the page so looking under `#idea_xx .idea-form` should be redundant.

File: app/assets/javascripts/ideas.js.coffee [hunk 3]
```diff
@@ -1,19 +1,34 @@
...
+        $("#idea-#{id} .idea-form").find('#idea_description').val(description)
+
```
Comment (swalberg) [inline app/assets/javascripts/ideas.js.coffee:33]: I'm seeing this  `$("#idea-#{id} something")` pattern here in every query. Is there a way to remove that redundancy and scope everything to `#idea-xx`?

File: app/assets/stylesheets/ideas.css.scss [hunk 4]
```diff
@@ -16,4 +16,24 @@
       margin-left: 10px;
     }
   }
+}
+
+.idea {
+
+  .edit {
+    @extend .btn;
```
Comment (swalberg) [inline app/assets/stylesheets/ideas.css.scss:24]: Aha, I like this pattern better than using the bootstrap classes within the HTML. Nice approach.

**Comment by amirci** [comment]
Please, try to remove the bin folder and bundle again...

**Comment by jeffreyfultonca** [comment]
I believe the binstubs generation was caused by having "BUNDLE_BIN: bin" in my .bundle/config. I have removed this line and am no longer generating the bin/ folder when launching rails server.

**Pull request closed without merging**